Python/Product/Uap/Interpreter/PythonUapInterpreter.cs

```diff
@@ -0,0 +1,237 @@
+﻿/* ****************************************************************************
+ *
+ * Copyright (c) Microsoft Corporation. 
+ *
+ * This source code is subject to terms and conditions of the Apache License, Version 2.0. A 
+ * copy of the license can be found in the License.html file at the root of this distribution. If 
+ * you cannot locate the Apache License, Version 2.0, please send an email to 
+ * vspython@microsoft.com. By using this source code in any fashion, you are agreeing to be bound 
+ * by the terms of the Apache License, Version 2.0.
+ *
+ * You must not remove this notice, or any other, from this software.
+ *
+ * ***************************************************************************/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PythonTools.Analysis;
+using Microsoft.PythonTools.Interpreter;
+using Microsoft.VisualStudioTools;
+
+namespace Microsoft.PythonTools.Uap.Interpreter {
+    class PythonUapInterpreter : IPythonInterpreter, IPythonInterpreterWithProjectReferences2, IDisposable {
```

Comment:
Other than preventing references (which would be better done by removing `IPythonInterpreterWithProjectReferences2`), what value does this class add over the default interpreter class? If you remove the `MakeInterpreter` override in the factory then you'll get the default for free.